v2/spiffetls/tlsconfig/config_test.go

```diff
@@ -0,0 +1,687 @@
+		},
+		{
+			name:       "wrapped is called",
+			authorizer: tlsconfig.AuthorizeAny(),
+			bundle:     bundle1,
+			err:        "wrapped called",
+			raw:        svid1Raw,
+			wrapped:    wrapped,
+		},
+	}
+
+	for _, testCase := range testCases {
+		testCase := testCase
+		t.Run(testCase.name, func(t *testing.T) {
+			verifyPeerCertificate := tlsconfig.WrapVerifyPeerCertificate(testCase.wrapped, testCase.bundle, testCase.authorizer)
+			require.NotNil(t, verifyPeerCertificate)
+
+			err := verifyPeerCertificate(testCase.raw, [][]*x509.Certificate{})
+			if testCase.err != "" {
+				require.EqualError(t, err, testCase.err)
+				return
+			}
+
+			require.NoError(t, err)
+		})
+	}
+}
+
+func TestTlsHandshake(t *testing.T) {
+	// Create Bundle1
+	td := spiffeid.RequireTrustDomainFromString("domain1.test")
+	ca1 := test.NewCA(t)
+	bundle1 := ca1.Bundle(td)
+	require.NotNil(t, bundle1)
+
+	// Create SVID
+	svid1ID := td.NewID("server")
+	svid1Certs, key1 := ca1.CreateX509SVID(svid1ID.String())
+	require.NotNil(t, svid1Certs)
+	require.NotNil(t, key1)
+	serverSVID := &x509svid.SVID{
+		ID:           svid1ID,
+		Certificates: svid1Certs,
+		PrivateKey:   key1,
+	}
+
+	// Create Bundle2
+	td2 := spiffeid.RequireTrustDomainFromString("domain2.test")
+	ca2 := test.NewCA(t)
+	bundle2 := ca2.Bundle(td2)
+	require.NotNil(t, bundle2)
+
+	testCases := []struct {
+		name         string
+		serverConfig *tls.Config
+		clientConfig *tls.Config
+		err          string
+	}{
+		{
+			name:         "success",
+			serverConfig: tlsconfig.TLSServerConfig(serverSVID),
+			clientConfig: tlsconfig.TLSClientConfig(bundle1, tlsconfig.AuthorizeAny()),
+		},
+		{
+			name:         "invalid trust domain",
+			serverConfig: tlsconfig.TLSServerConfig(serverSVID),
+			clientConfig: tlsconfig.TLSClientConfig(bundle1, tlsconfig.AuthorizeMemberOf(td2)),
+			err:          `unexpected trust domain "domain1.test"`,
+		},
+		{
+			name:         "invalid bundle",
+			serverConfig: tlsconfig.TLSServerConfig(serverSVID),
+			clientConfig: tlsconfig.TLSClientConfig(bundle2, tlsconfig.AuthorizeMemberOf(td)),
+			err:          `x509svid: could not get X509 bundle: x509bundle: no X.509 bundle found for trust domain: "domain1.test"`,
+		},
+	}
+
+	for _, testCase := range testCases {
+		testCase := testCase
+		t.Run(testCase.name, func(t *testing.T) {
+			testConnection(t, testCase.serverConfig, testCase.clientConfig, testCase.err)
+		})
+	}
+}
+
+func TestMTLSHandshake(t *testing.T) {
+	// Create Bundle1
+	td := spiffeid.RequireTrustDomainFromString("domain1.test")
+	ca1 := test.NewCA(t)
+	bundle1 := ca1.Bundle(td)
+	require.NotNil(t, bundle1)
+
+	// Create Server SVID
+	svid1ID := td.NewID("server")
+	svid1Certs, key1 := ca1.CreateX509SVID(svid1ID.String())
+	require.NotNil(t, svid1Certs)
+	require.NotNil(t, key1)
+	serverSVID := &x509svid.SVID{
+		ID:           svid1ID,
+		Certificates: append(svid1Certs, ca1.Roots()...),
+		PrivateKey:   key1,
+	}
+
+	// Create Client SVID
```

Comment:
did you intend to leave this commented out?

Reply:
yes